pywhy_graphs/algorithms/generic.py

```diff
@@ -333,3 +334,236 @@ def _single_shortest_path_early_stop(G, firstlevel, paths, cutoff, join, valid_p
+
+    Returns
+    -------
+    iscollider : bool
+        Bool is set true if the node is a collider, false otherwise.
+    """
+    parents = _directed_sub_graph_parents(G, cur_node)
+    parents = parents.union(_bidirected_sub_graph_neighbors(G, cur_node))
+
+    if prev_node in parents and next_node in parents:
+        return True
+
+    return False
+
+
+def _recursive_path(G, node_x, node_y, L, S, visited, check_ancestors, cur_node, prev_node):
+    """Recursively explores a graph to find a path.
+
+       Finds path that are compliant with the inducing path requirements.
+
+    Parameters
+    ----------
+    G : graph
+        The graph.
+    node_x : node
+        The source node.
+    node_y : node
+        The destination node
+    L : set
+        Set containing all the non-colliders.
+    S : set
+        Set containing all the colliders.
+    visited : set
+        Set containing all the nodes already visited.
+    check_ancestors : set
+        Set containing all the ancestors to be checked against.
+    cur_node : node
+        The current node.
+
+    Returns
+    -------
+    path : Tuple[bool, path]
+        A tuple containing a bool and a path which is empty if the bool is false.
+    """
+    path_exists = False
+    path = []
+    visited.add(cur_node)
+    children = G.neighbors(cur_node)
+
+    if cur_node is node_y:
+        return (True, [node_y])
+
+    for elem in children:
+        if elem in visited:
+            continue
+
+        else:
+            if (
+                _is_collider(G, prev_node, cur_node, elem)
+                and (cur_node not in check_ancestors)
+                and (cur_node not in S)
+                and (cur_node is not node_y)
+            ):
+                continue
+
+            elif (
+                not _is_collider(G, prev_node, cur_node, elem)
+                and (cur_node not in L)
+                and (cur_node is not node_y)
+            ):
+                continue
+
+            path_exists, temp_path = _recursive_path(
+                G, node_x, node_y, L, S, visited, check_ancestors, elem, cur_node
+            )
+
+            if path_exists:
+                path.append(cur_node)
+                path.extend(temp_path)
+                break
+
+    return (path_exists, path)
+
+
+def inducing_path(G, node_x, node_y, L=None, S=None):
+    """Checks if an inducing path exists between two nodes as defined in :footcite:`Zhang2008`.
+
+    Parameters
+    ----------
+    G : Graph
+        The graph.
+    node_x : node
+        The source node.
+    node_y : node
+        The destination node.
+    L : set
+        Nodes that are ignored on the path. Defaults to an empty set. See Notes for details.
+    S:  set
+        Nodes that are always conditioned on. Defaults to an empty set. See Notes for details.
+
+    Returns
+    -------
+    path : Tuple[bool, path]
+        A tuple containing a bool and a path if the bool is true, an empty list otherwise.
+
+    Notes
```

Comment:
Can I also trouble you to write a minimal working example? e.g. see networkx/numpy/scikit-learn documentation

Reply:
@adam2392 Where do you want me to add the example? And what format should it be in? Should it be a Python code with comments explaining what's happening or an article with Python snippets?

Reply:
Can I add the example to the plot_mixed_edge_graph.ipynb? Would that be automatically included in the docs?

Reply:
You can add one in the examples directory as a Python script. 

You can see some of the patterns used for rendering the example and how it's done in the other examples. 

We basically use Sphinx to render this into a notebook which is ran on the CI pipelines.

Reply:
Alright, let me try and write a first draft.